### PR TITLE
Service.cors_supported_headers are now filtered by method

### PR DESCRIPTION
**Pull request by leplatrem**
Fix proposal of https://github.com/mozilla-services/cliquet/issues/54

*caution*, this is a violent change, breaking the Service API (from property to class method)

### CODE REVIEW

**Comment by leplatrem** [comment]
I suggest we also change ``cors_supported_methods`` and ``cors_supported_origins`` to be methods instead of properties.

File: cornice/service.py [hunk 1]
```diff
@@ -413,17 +413,21 @@ def cors_enabled(self):
     def cors_enabled(self, value):
         self._cors_enabled = value
 
-    @property
-    def cors_supported_headers(self):
+    def cors_supported_headers(self, method=None):
```
Comment (ametaireau) [inline cornice/service.py]: When exposing functions with the first argument `method`, we've been following the `{methodname}_for` paradigm, maybe we should stick with this.

Comment (ametaireau) [inline cornice/service.py]: Also, that would let us be backwards compatible, exposing the properties to the people who where already using them. In this case, we should issue a deprecation warning and remove it in a later release.

**Comment by ametaireau** [comment]
Otherwise, +1 with the idea.

File: cornice/service.py (hunk 1 above)
Comment (leplatrem) [inline cornice/service.py]: I updated the PR, and generalized it to cors_support_credentials.

It looks weird when method param is omitted, but it's now consistent

**Comment by leplatrem** [comment]
@ametaireau I now raise DeprecationWarning for previous properties/methods. Does it suit what you had in mind ?

**Comment by leplatrem** [comment]
Thanks for your feedback, I changed the deprecation warnings to fit existing code in Cornice

**Comment by ametaireau** [comment]
thanks!

**Pull request merged**